### ISSUE TITLE
Curvature CI + flatness test (#944) only in the HTML report: wire model.curvature_ci() / model.flatness_test()

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#1059]
**Part of audit #1052.** Programmatic-API gap on top of #944.

### Feature implemented but only reachable as a report side-effect
`src/geometry/curvature_estimand.rs` (#944) implements the constant-curvature estimand with exact-REML κ̂, profile confidence intervals, and a flatness LR test: `profile_ci_walk`, `flatness_lr_test`, `KappaProfileCi`, `FlatnessTest`, `wald_half_width`, `DesignCoordKappaJet`. The CI/test are computed **only inside `src/terms/smooth.rs`** for the HTML report — grep shows no other caller and no FFI/Python getter.

### Missing seam
A user fitting a manifold smooth gets κ̂ and its CI rendered in the report but **cannot retrieve them programmatically** (no `model.curvature_ci(...)`). The flatness test (is the manifold significantly curved vs flat?) is invisible to code.

### Proposed magic-by-default design
- `model.curvature_ci(term, alpha=0.05)` → `KappaProfileCi` (point estimate + profile interval).
- `model.flatness_test(term)` → `FlatnessTest` (LR stat, p-value for H0: κ=0).
Seam: surface the already-computed `KappaProfileCi`/`FlatnessTest` from the smooth term's report struct through a pyffi getter keyed by term index. No recomputation.

**Comment by SauersML** [comment]
Investigated #1059. **The programmatic seam is already fully wired** (landed in prior #944/#1063 work) — this issue is effectively complete on the wiring axis:

- Rust core: `gam::smooth::curvature_inference_forspec` (src/terms/smooth.rs:21510) computes κ̂ + profile CI + interior κ=0 flatness LR via `crate::geometry::curvature_estimand`.
- FFI: `curvature_inference_json` (crates/gam-pyffi/src/lib.rs:17162, impl 28278) is registered in the module (line 23622) and re-profiles V_p(κ) on the training data.
- Python: `Model.curvature(data, level=0.95)` (gamfit/_model.py:336) returns kappa_hat, ci_lo/ci_hi, lo/hi_at_bound, verdict, flatness_lr_stat, flatness_p_value. `Model.summary().curvature_estimands` surfaces κ̂ with zero refit.

The implemented design uses a single `model.curvature(...)` returning the full report (a superset of the proposed `curvature_ci()`/`flatness_test()` — one call, no recomputation), which satisfies the magic-by-default intent.

**Test fix landed:** the Rust e2e `constant_curvature_kappa_inference_e2e.rs` was failing at term-spec build — `termspec_for` built the schema dataset from a 1-row zeros placeholder, which `build_smooth_basis`'s degenerate-column guard correctly rejects (smooth term over x1 has only one unique value). Fixed to hand the real `[y,x1,x2]` frame to the spec builder. The path now fits, infers, and returns **finite, well-formed** κ̂/CI/flatness (issue acceptance: "returned and finite" — met).

**Separate #944 fit bug newly exposed (out of #1059 scope, flagging not weakening):** with the spec-build fixed, the κ-as-outer-ψ optimizer **rails κ̂ to the +2.0 chart bound for all three truths** (κ⋆ = −2, 0, +2 all return κ̂ ≈ +2.003, CI pinned at the bound, flatness always rejected). κ is not identified by the profiled REML criterion here — V_p(κ) appears monotone toward the positive bound regardless of data. Suspects: the Frobenius-normalized RKHS penalty (`normalize_penalty` in constant_curvature_smooth.rs) decoupling the logdet from κ, or a sign in `build_constant_curvature_basis_kappa_derivatives`. This is a #944 stage-3 fit-internals defect, not an inference-surfacing gap. The truth-recovery assertions in the e2e are left **honestly red** (no weakening) to track it; recommend a dedicated #944 follow-up issue.

**Comment by SauersML** [comment]
NOT fixed — verified RED on MSI (HEAD dede6c4fe): `cargo test --test constant_curvature_kappa_inference_e2e`.

`vp_grid_identifies_planted_kappa_sign` FAILS: for **hyperbolic (κ<0) truth**, the profiled REML criterion V_p(κ) is **monotonically decreasing toward the +κ bound** and the argmin rails to +1.90 regardless of the planted negative curvature:
```
κ=-1.90  V_p=122.807
κ=+0.00  V_p=119.445
κ=+1.00  V_p=115.401
κ=+1.90  V_p=110.232   <- argmin
```
"hyperbolic: V_p must be minimised at κ<0 for hyperbolic truth, got argmin=1.9" (line 247). The hyperbolic_/spherical_truth_recovers_* arms inherit the same non-identification.

The programmatic inference/surfacing seam (model.curvature(...)) is wired and returns finite κ̂/CI/flatness — that part is done. But the underlying **#944 stage-3 fit defect remains**: κ is not identified by V_p (the criterion is monotone in κ, not convex around the truth), so the κ̂ point estimate and sign are wrong. The earlier fix attempt (628a7b46d) did not resolve it. Leaving open; this needs the #944 κ-derivative / penalty-normalization root cause, not an inference-surfacing change.
